examples/ChatQnA/deploy/k8s_helm.md

```diff
@@ -0,0 +1,497 @@
+```
+The above output has been parsed into the below sentence which shows how the LLM has picked up the right context to answer the question correctly after the document upload:
+```
+OPEN Platform for Enterprise AI (Open Platform for Enterprise AI) is a framework that focuses on creating and evaluating open, multi-provider, robust, and composable generative AI (GenAI) solutions. It aims to facilitate the implementation of enterprise-grade composite GenAI solutions, particularly Retrieval Augmented Generative AI (RAG), by simplifying the integration of secure, performant, and cost-effective GenAI workflows into business systems.
+```
+
+### TEI Embedding Service
+Use the following command to forward traffic from your local machine to the service running in the Kubernetes cluster:
+```bash
+kubectl port-forward svc/chatqna-tei 6006:80
+```
+Follow the below steps in a different terminal.
+
+The TEI embedding service takes in a string as input, embeds the string into a
+vector of a specific length determined by the embedding model and returns this
+embedded vector.
+
+```
+curl http://localhost:6006/embed \
+    -X POST \
+    -d '{"inputs":"What is Deep Learning?"}' \
+    -H 'Content-Type: application/json'
+```
+
+In this example the embedding model used is "BAAI/bge-base-en-v1.5", which has a vector size of 768. So the output of the curl command is a embedded vector of
+length 768.
+
+
+### Retriever Microservice
+Use the following command to forward traffic from your local machine to the service running in the Kubernetes cluster:
+```bash
+kubectl port-forward svc/chatqna-retriever-usvc 7000:7000
+```
+Follow the below steps in a different terminal.
+
+To consume the retriever microservice, you need to generate a mock embedding
+vector by Python script. The length of embedding vector is determined by the
+embedding model. Here we use the
+model EMBEDDING_MODEL_ID="BAAI/bge-base-en-v1.5", which vector size is 768.
+
+Check the vector dimension of your embedding model and set
+`your_embedding` dimension equal to it.
+
+```
+export your_embedding=$(python3 -c "import random; embedding = [random.uniform(-1, 1) for _ in range(768)]; print(embedding)")
```

Comment:
Both `export` and `embedding` var are redundant:
```suggestion
your_embedding=$(python3 -c "import random; print([random.uniform(-1, 1) for _ in range(768)])")
```

Reply:
@eero-t I believe it's more readable this way for the user, and the source for this is the dev team's readme.